src/runtime_src/core/common/dev_factory.h

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+#ifndef _XCL_DEV_FACTORY_H_
+#define _XCL_DEV_FACTORY_H_
+
+#include "device.h"
+#include <vector>
+
+namespace xrt_core {
+
+// One function on FPGA or AIE device
+class dev
+{
+public:
+  bool m_is_mgmt = false;
+  bool m_is_ready = true; 
+  dev(bool isuser) {  
+      m_is_mgmt = !isuser;  
+  }
+  
+  virtual ~dev() {}
+ 
+  // Hand out a "device" instance that is specific to this type of device.
+  // Caller will use this device to access device specific implementation of ishim.
+  virtual std::shared_ptr<device>
+  create_device(device::handle_type handle, device::id_type id) const = 0;
+
+  // Hand out an opaque "shim" handle that is specific to this type of device.
+  // On legacy Alveo device, this handle can be used to lookup a device instance and
+  // make xcl HAL API calls.
+  // On new platforms, this handle can only be used to look up a device. HAL API calls
+  // through it are not supported any more.
+  virtual device::handle_type
+  create_shim(device::id_type id) const = 0;
+
+  //get BDF 
+  virtual std::tuple<uint16_t, uint16_t, uint16_t, uint16_t>
+  get_bdf_info() const
+  {
+    return std::make_tuple(uint16_t(0), uint16_t(0), uint16_t(0), uint16_t(0));
+  }
+};
+
+
+size_t
+get_dev_ready(bool user = true);
+
+size_t
+get_dev_total(bool user = true);
```

Comment:
`get_device_ready` and `get_device_total`  and comment what this means.

Reply:
sure